src/transformers/models/vits/modeling_vits.py

```diff
@@ -430,6 +505,131 @@ def forward(self, inputs, padding_mask, global_conditioning=None):
         sampled = (mean + torch.randn_like(mean) * torch.exp(log_stddev)) * padding_mask
         return sampled, mean, log_stddev
 
+    def apply_weight_norm(self):
+        self.wavenet.apply_weight_norm()
+
+    def remove_weight_norm(self):
+        self.wavenet.remove_weight_norm()
+
+    def resize_speaker_embeddings(self, speaker_embedding_size: Optional[int] = None):
+        self.wavenet.speaker_embedding_size = speaker_embedding_size
+        hidden_size = self.wavenet.hidden_size
+        num_layers = self.wavenet.num_layers
+
+        cond_layer = torch.nn.Conv1d(speaker_embedding_size, 2 * hidden_size * num_layers, 1)
+        self.wavenet.cond_layer = nn.utils.weight_norm(cond_layer, name="weight")
+        nn.init.kaiming_normal_(self.wavenet.cond_layer.weight)
+        if self.wavenet.cond_layer.bias is not None:
+            k = math.sqrt(
+                self.wavenet.cond_layer.groups
+                / (self.wavenet.cond_layer.in_channels * self.wavenet.cond_layer.kernel_size[0])
+            )
+            nn.init.uniform_(self.wavenet.cond_layer.bias, a=-k, b=k)
```

Comment:
Here I think we should be using self._init_weights(cond_layer). Is the logic different from the weight initialization in the pretrained model?

Reply:
You're right, good catch!